src/rmp/src/Restructure.cpp

```diff
@@ -156,20 +161,25 @@ void Restructure::runABC()
 
   // abc optimization
   bool area_mode = opt_mode_ <= Mode::AREA_3;
-  std::string abc_script_file = work_dir_name_ + "ord_abc_script.tcl";
   std::string best_blif;
   int best_inst_count = std::numeric_limits<int>::max();
-  for (int opt_mode = static_cast<int>(area_mode)
+  int opt_mode_init = static_cast<int>(area_mode)
```

Comment:
why is area_mode being case to an int - it is just being used as a boolean expression and is already a bool.

Reply:
way too many casts in this code.

Reply:
@jjcherry56 Iterating enums in c/c++ has always sucked.   I've thought about importing something like better-enum or magic-enum in the past.  How do you think this should be improved?

Reply:
Well, I agree they suck, which is why I defined EnumNameMap.hh in opensta.
I am not much of a fan of using the preprocessor like you did in Logger.h, but it beats nothing.

Reply:
How does that solve iteration?

Reply:
As it stands today it doesn't but adding a begin()/end() is all it would take for range iteration to work.
The code in question should be using array iteration on an array initialized with the enum values.
As it stands it is incredibly messy mixing 2 different ranges in the same loop and assuming a specific
order for the enum values.

Reply:
The difference in mode counts complicates this but something like this:
if (area_mode) {
    std::array<Mode, 3> area_modes{Mode::AREA_1, Mode::AREA_2, Mode::AREA_3};
    loop_body(area_modes);
  } else {
    std::array<Mode, 4> delay_modes{Mode::DELAY_1, Mode::DELAY_2, Mode::DELAY_3, Mode::DELAY_4};    
    loop_body(delay_modes);
  }